source/frontend/src/features/leases/detail/LeasePages/improvements/Improvements.tsx

```diff
@@ -13,6 +15,21 @@ export const Improvements: React.FunctionComponent<IImprovementsProps> = props =
   const { values } = useFormikContext<IFormLease>();
   const improvements: ILeaseImprovement[] = getIn(values, 'improvements') ?? [];
 
+  const { getByType } = useLookupCodeHelpers();
+  const improvementTypeCodes = getByType(API.PROPERTY_IMPROVEMENT_TYPES);
+
+  improvements.sort((improvementOne: ILeaseImprovement, improvementTwo: ILeaseImprovement) => {
+    const findDisplayOrder = (x: ILeaseImprovement): number => {
+      for (let typeCode of improvementTypeCodes) {
```

Comment:
this is find but the .find javascript function would provide a more concise solution:
ie:
improvementTypeCodes.find(itc => itc.id === x.propertyImprovementTypeId)

also, be careful naming a variable x. If you start to have multiple variables named like that in scope at the same time reading the code can be very confusing:
bad: x
better: li
best: leaseImprovement

Recommend using the full proper english name unless length is a concern or the logic is very simple. If length is a concern that may be code smell to refactor/simplify the logic you are working on.